File: mobile/lib/model/rent.dart

class Rent{
  final String id;
  final String title;
  final String owner;
  final String description;
  final String image;
  final int size;
  final int room_number;
  final String location;
  final num price_per_month;
  final String start_date_available;
  final String end_date_available;

  const Rent({
    required this.id,
    required this.title,
    required this.owner,
    required this.description,
    required this.image,
    required this.size,
    required this.room_number,
    required this.location,
    required this.price_per_month,
    required this.start_date_available,
    required this.end_date_available,
  });
  factory Rent.fromJson(Map<String, dynamic> json) {
    return Rent(
        id: json['id'],
        title: json['title'],
        owner: json['owner'],
        description: json['description'],
        image: json['image'],
        size: json['size'],
        room_number: json['room_number'],
        location: json['location'],
        price_per_month: json['price_per_month'],
        start_date_available: json['start_date_available'],
        end_date_available: json['end_date_available']
    );
  }

}

File: Backend/models/rent.js
const mongoose = require('mongoose');

const rentSchema = new mongoose.Schema({
    title: { type: String, required: true },
    owner: { type: String, required: true },
    description: { type: String, required: true },
    image: { type: String, required: true },
    size: { type: Number, required: true },
    room_number: { type: Number, required: true },
    location: { type: String, required: true },
    price_per_month: { type: Number, required: true },
    start_date_available: { type: String, required: true },
    end_date_available: { type: String, required: true },
    user_id: { type: String, required: true }
});

module.exports = mongoose.model('Rent', rentSchema);